scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/TokenClassifiers.scala

```diff
@@ -5,64 +5,6 @@ import scala.meta.internal.classifiers.classifier
 import scala.meta.tokens.Token
 import scala.meta.tokens.Token._
 
-@classifier
-trait CloseDelim {
-  def unapply(token: Token): Boolean = {
-    token.is[RightBrace] || token.is[RightBracket] || token.is[RightParen] ||
-    token.is[Indentation.Outdent]
-  }
-}
-
-@classifier
-trait Literal {
-  def unapply(token: Token): Boolean = token match {
-    case Constant.Int(_) => true
-    case Constant.Long(_) => true
-    case Constant.Float(_) => true
-    case Constant.Double(_) => true
-    case Constant.Char(_) => true
-    case Constant.Symbol(_) => true
-    case Constant.String(_) => true
-    case KwTrue() => true
-    case KwFalse() => true
-    case KwNull() => true
-    case _ => false
-  }
-}
-
-@classifier
-trait NumericLiteral {
-  def unapply(token: Token): Boolean = token match {
-    case Constant.Int(_) => true
-    case Constant.Long(_) => true
-    case Constant.Float(_) => true
-    case Constant.Double(_) => true
-    case _ => false
-  }
-}
-
-@classifier
-trait Whitespace {
-  def unapply(token: Token): Boolean = {
-    token.is[Space] || token.is[Tab] ||
-    token.is[LineEnd] || token.is[FF]
-  }
-}
-
-@classifier
-trait LineEnd {
```

Comment:
this is what I called `AtEOL`, it's used in a couple of places around the codebase.